pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -367,6 +368,9 @@ func (cl *Client) addCRD(name string) {
 		ObjectFilter:    composeFilters(namespaceFilter, cl.inRevision, extraFilter),
 		ObjectTransform: transform,
 	}
+	if resourceGVK == gvk.Gateway || resourceGVK == gvk.KubernetesGateway {
```

Comment:
This is suspicious. Why not just KubernetesGateway

Reply:
Because Istio Ingress Gateways can also be labeled with tags.

Reply:
Good point, will remove.  And gateway injection uses mwc, which already respects tags.